src/components/layout/Header.js

```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import Logo from "../logo";
+import MenuItem from "./MenuItem";
+
+import { datocmsMenuItemTypes } from "./proptypes";
+import { prepareMenuItems } from "./utils";
+
+const Header = ({ menuData }) => {
+  const menuItems = prepareMenuItems(menuData, "header");
```

Comment:
```suggestion
  const menuItems = menuData.filter(item => item.showInMainMenu)
```

Зачем городить этот хелпер и режимы и так далее? Вот - простой и понятный код, который делает то что надо. Так же в футере.

Reply:
теперь Header и Footer сами запрашивают необходимые данные с фильтрами